### PR TITLE
Optional enable compression middleware to reduce response size  (disabled by default)

### PR DESCRIPTION
**Pull request by zuisong**
#523 
#497 

This PR adds an optional section to turn on compress response, turned off by default, and a note in the help docs about the impact of turning this on.

```
/// Enable compress response
///
/// WARNING: Enabling this option may slow down transfers due to CPU overhead, so it is disabled by default.
/// 
/// Only enable this option if you know that your users have slow connections or if you want to minimize your server's bandwidth usage.
```

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -202,6 +203,10 @@ async fn run(miniserve_config: MiniserveConfig) -> Result<(), ContextualError> {
             .app_data(stylesheet.clone())
             .wrap_fn(errors::error_page_middleware)
             .wrap(middleware::Logger::default())
+            .wrap(Condition::new(
+                miniserve_config.compress_response_enabled,
```
Comment (svenstaro) [inline src/main.rs:207]: Let's be consistent with ourselves here and remove the extra import at the top.
```suggestion
            .wrap(middleware::Condition::new(
                miniserve_config.compress_response_enabled,
```

**Review by svenstaro — CHANGES_REQUESTED** [review]
Thanks! Just a tiny code change.

Did you perform some benchmarks by any chance?

**Comment by zuisong** [comment]
I didn't do benchmark, but my server egress bandwidth is very low, and I'm transmitting plain text files, which really need compression.

File: src/args.rs [hunk 2]
```diff
@@ -214,6 +214,17 @@ pub struct CliArgs {
...
+    ///
+    /// WARNING: Enabling this option may slow down transfers due to CPU overhead, so it is disabled by default.
+    ///
+    /// Only enable this option if you know that your users have slow connections or if you want to minimize your server's bandwidth usage.
+    #[arg(
+        long = "enable-compress-response",
+        env = "MINISERVE_ENABLE_COMPRESS_RESPONSE"
+    )]
+    pub enable_compress_response: bool,
+
```
Comment (svenstaro) [inline src/args.rs:227]: What do you think about just calling this `--compress-response`? The `--enable` prefixes are only there for archiving. Also, let's also give it `short = 'C'`.

**Review by svenstaro — CHANGES_REQUESTED** [review]

File: src/args.rs (hunk 2 above)
Comment (zuisong) [reply src/args.rs]: I think that's a great idea. I've actually considered doing something like that before.

**Comment by svenstaro** [comment]
Super nice, thanks!

**Pull request merged**